### PR TITLE
fix(ui): Add missing Authenticator translations

### PR DESCRIPTION
**Pull request by joebuono**
#### Description of changes

- Adds `AuthenticatorDictionary` TypeScript type
- Adds missing translations to all dictionaries, provided by Totoro
- Separates additional translations provided by customers
- Alphabetizes all translations

#### Description of how you validated changes

Ran yarn test

#### Checklist

- [X] PR description included
- [X] `yarn test` passes

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/ui/src/i18n/dictionaries/authenticator/it.ts [hunk 1]
```diff
@@ -1,52 +1,78 @@
...
   'Enter your username': 'Inserisci il tuo nome utente',
-  'Forgot Password': 'Password dimenticata',
   'Forgot your password?': 'Password dimenticata?',
+  'Hide password': 'Nascondi password',
+  Loading: 'Caricamento in corso',
+  'New password': 'Nuova password',
+  Password: 'Password',
+  'Phone Number': 'Numero di telefono',
+  'Resend Code': 'Rispedisci codice',
+  'Reset your Password': 'Reimposta la password',
```
Comment (ioanabrooks) [inline packages/ui/src/i18n/dictionaries/authenticator/it.ts:32]: This should be `Reimposta la tua password`

File: packages/ui/src/i18n/dictionaries/authenticator/it.ts [hunk 2]
```diff
@@ -1,52 +1,78 @@
...
+  'Reset your password': 'Reimposta la tua password',
```
Comment (ioanabrooks) [inline packages/ui/src/i18n/dictionaries/authenticator/it.ts:33]: And this should be `Reimposta la tua Password`

File: packages/ui/src/i18n/dictionaries/authenticator/de.ts [hunk 3]
```diff
@@ -1,68 +1,86 @@
...
   'Forgot your password?': 'Passwort vergessen? ',
-  'Forgot Password': 'Passwort vergessen',
   'Hide password': 'Passwort verbergen',
+  Loading: 'Wird geladen',
+  'New password': 'Neues Passwort',
+  Password: 'Passwort',
+  'Phone Number': 'Telefonnummer',
+  'Resend Code': 'Code erneut senden',
+  'Reset your Password': 'Passwort zurücksetzen',
+  'Reset your password': 'Zurücksetzen des Passworts',
```
Comment (ioanabrooks) [inline packages/ui/src/i18n/dictionaries/authenticator/de.ts:33]: `Passworts` is plural, should lines 32-33 be 
```
 'Reset your Password': 'Zurücksetzen des Passwort',
 'Reset your password': 'Zurücksetzen des passwort',
```

File: packages/ui/src/i18n/dictionaries/authenticator/de.ts [hunk 4]
```diff
@@ -1,68 +1,86 @@
...
+  'Enter your username': 'Geben Sie Ihren Benutzernamen ein',
   'Forgot your password?': 'Passwort vergessen? ',
-  'Forgot Password': 'Passwort vergessen',
   'Hide password': 'Passwort verbergen',
+  Loading: 'Wird geladen',
+  'New password': 'Neues Passwort',
+  Password: 'Passwort',
+  'Phone Number': 'Telefonnummer',
+  'Resend Code': 'Code erneut senden',
+  'Reset your Password': 'Zurücksetzen des Passwort',
```
Comment (ErikCH) [inline packages/ui/src/i18n/dictionaries/authenticator/de.ts:32]: These did not change any of the user submitted translations correct?

Comment (joebuono) [reply packages/ui/src/i18n/dictionaries/authenticator/de.ts:32]: Correct

**Review by ioanabrooks — APPROVED** [review]

**Review by ErikCH — APPROVED** [review]

**Pull request merged**